File: unit-2/sprint-2/day-2-RecyclerView/we/RecyclerView/app/src/main/java/com/example/recyclerview/MainActivity.java
package com.example.recyclerview;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity  {

    private RecyclerView recyclerView;
    private ArrayList<Student> studentList;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        recyclerView = findViewById(R.id.recyclerView);
        buildList();
        setRecyclerView();
    }

    private void buildList() {
        studentList = new ArrayList<>();
        for (int i = 0; i < 15; i++) {
            studentList.add(new Student("Nishant", 24, ""+i));
         }
    }
    private void setRecyclerView(){
        StudentAdapter adapter = new StudentAdapter(studentList);
        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this);
        recyclerView.setLayoutManager(linearLayoutManager);
        recyclerView.setAdapter(adapter);
    }
}

File: MVVM Test/app/src/main/java/com/example/mvvmtest/ui/MainActivity.kt
package com.example.mvvmtest.ui

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.mvvmtest.MainUIModel
import com.example.mvvmtest.R
import com.example.mvvmtest.data.models.ResultsDTO
import com.example.mvvmtest.ui.adapter.MainAdapter
import com.example.mvvmtest.ui.viewmodel.MainViewModel
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    private lateinit var mainViewModel: MainViewModel
    private var list = emptyList<ResultsDTO>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        mainViewModel = ViewModelProvider(this).get(MainViewModel::class.java)
        mainViewModel.callApi()
        mainViewModel.liveData.observe(this, {

            it.let {
                when (it) {
                    is MainUIModel.OnSuccess -> {
                        list = it.responseDTO.results as ArrayList<ResultsDTO>
                        setRecyclerAdapter()
                    }

                    is MainUIModel.onFailure -> {
                        Toast.makeText(this@MainActivity, it.error, Toast.LENGTH_SHORT).show()
                    }
                }
            }
        })

    }

    fun setRecyclerAdapter() {
        val mainAdapter = MainAdapter(list)
        val linearLayoutManager = LinearLayoutManager(this)
        recyclerview.apply {
            adapter = mainAdapter
            layoutManager = linearLayoutManager
        }
    }
}